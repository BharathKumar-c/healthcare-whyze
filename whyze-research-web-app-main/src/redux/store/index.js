import { configureStore } from '@reduxjs/toolkit';
import reducer from '../reducer/combinedReducer';

const store = configureStore({
  reducer: reducer,

  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: false,
    }),
});

export default store;
