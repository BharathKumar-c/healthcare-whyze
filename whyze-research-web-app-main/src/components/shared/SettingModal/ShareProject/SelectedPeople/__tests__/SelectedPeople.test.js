import { fireEvent, render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../../../../../../redux/store';
import React from 'react';
import SelectedPeoples from '..';
function renderWithContext(element) {
  const { container, getByText, getByRole } = render(
    <Provider store={store}>{element}</Provider>,
  );
  return { container, store, getByText, getByRole };
}

describe('create test cases for SelectedPeoples', () => {
  test('create snapshots for SelectedPeoples', () => {
    const { container } = renderWithContext(<SelectedPeoples />);
    expect(container).toMatchSnapshot();
  });
  test('should triggers checkbox', () => {
    const { container, getByRole } = renderWithContext(<SelectedPeoples />);
    const clickCheckBox = getByRole('checkbox');
    fireEvent.click(clickCheckBox);
    expect(container).toMatchSnapshot();
  });
  test('should check tagrender', () => {
    const { container } = renderWithContext(<SelectedPeoples />);
    const shareSelect = screen.getAllByRole('combobox');
    expect(shareSelect[0]).toBeInTheDocument();
    expect(shareSelect[0]).toHaveAttribute(
      'aria-label',
      'Add-people-ariaLabel',
    );
    fireEvent.click(shareSelect[0]);
    fireEvent.change(shareSelect[0], { target: { value: 'CR' } });

    const option = screen.getByTitle('CR');
    fireEvent.click(option);
    expect(shareSelect[0].value).toBe('');
    const tegRender = container.getElementsByClassName('tag-container');
    expect(tegRender[0]).toBeInTheDocument();
  });
});

// import React from 'react';
// import { render, fireEvent } from '@testing-library/react';
// import SelectedPeoples from '..';

// describe('SelectedPeoples', () => {

// });
