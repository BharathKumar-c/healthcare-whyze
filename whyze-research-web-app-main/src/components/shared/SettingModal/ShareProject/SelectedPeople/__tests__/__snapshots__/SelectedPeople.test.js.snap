// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create test cases for SelectedPeoples create snapshots for SelectedPeoples 1`] = `
<div>
  <div
    class="SelectedPeoples-container"
  >
    <div
      class="ant-row"
    >
      <div
        class="ant-col ant-col-19"
      >
        <div
          class="alignment"
        >
          <div
            aria-label="Add-people-ariaLabel"
            class="ant-select ant-select-multiple ant-select-show-search"
          >
            <div
              class="ant-select-selector"
            >
              <div
                class="ant-select-selection-overflow"
              >
                <div
                  class="ant-select-selection-overflow-item ant-select-selection-overflow-item-suffix"
                  style="opacity: 1;"
                >
                  <div
                    class="ant-select-selection-search"
                    style="width: 0px;"
                  >
                    <input
                      aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                      aria-autocomplete="list"
                      aria-controls="rc_select_TEST_OR_SSR_list"
                      aria-haspopup="listbox"
                      aria-label="Add-people-ariaLabel"
                      aria-owns="rc_select_TEST_OR_SSR_list"
                      autocomplete="off"
                      class="ant-select-selection-search-input"
                      id="rc_select_TEST_OR_SSR"
                      role="combobox"
                      type="search"
                      value=""
                    />
                    <span
                      aria-hidden="true"
                      class="ant-select-selection-search-mirror"
                    >
                       
                    </span>
                  </div>
                </div>
              </div>
              <span
                class="ant-select-selection-placeholder"
              >
                Add people and groups
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-col ant-col-4"
      >
        <div
          class="alignment"
        >
          <div
            class="ant-select condition-container-custom-select ant-select-single ant-select-show-arrow"
          >
            <div
              class="ant-select-selector"
            >
              <span
                class="ant-select-selection-search"
              >
                <input
                  aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                  aria-autocomplete="list"
                  aria-controls="rc_select_TEST_OR_SSR_list"
                  aria-haspopup="listbox"
                  aria-owns="rc_select_TEST_OR_SSR_list"
                  autocomplete="off"
                  class="ant-select-selection-search-input"
                  id="rc_select_TEST_OR_SSR"
                  readonly=""
                  role="combobox"
                  style="opacity: 0;"
                  type="search"
                  unselectable="on"
                  value=""
                />
              </span>
              <span
                class="ant-select-selection-item"
                title="Viewer"
              >
                Viewer
              </span>
            </div>
            <span
              aria-hidden="true"
              class="ant-select-arrow"
              style="user-select: none;"
              unselectable="on"
            >
              <span
                aria-label="down"
                class="anticon anticon-down ant-select-suffix"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="down"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                  />
                </svg>
              </span>
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      class="SelectedPeoples-container_checkbox"
    >
      <div>
        <label
          class="ant-checkbox-wrapper"
        >
          <span
            class="ant-checkbox"
          >
            <input
              class="ant-checkbox-input"
              name="Notify people"
              type="checkbox"
              value=""
            />
            <span
              class="ant-checkbox-inner"
            />
          </span>
        </label>
        <span
          class="ant-typography"
          style="margin-left: 1rem;"
        >
          Notify people
        </span>
      </div>
      <div
        class="SelectedPeoples-container_textArea"
        id=""
      >
        <span
          class="ant-typography text-area-lable"
        >
          Message
        </span>
        <textarea
          class="ant-input text-area-input"
          name=""
          placeholder=""
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`create test cases for SelectedPeoples should triggers checkbox 1`] = `
<div>
  <div
    class="SelectedPeoples-container"
  >
    <div
      class="ant-row"
    >
      <div
        class="ant-col ant-col-19"
      >
        <div
          class="alignment"
        >
          <div
            aria-label="Add-people-ariaLabel"
            class="ant-select ant-select-multiple ant-select-show-search"
          >
            <div
              class="ant-select-selector"
            >
              <div
                class="ant-select-selection-overflow"
              >
                <div
                  class="ant-select-selection-overflow-item ant-select-selection-overflow-item-suffix"
                  style="opacity: 1;"
                >
                  <div
                    class="ant-select-selection-search"
                    style="width: 0px;"
                  >
                    <input
                      aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                      aria-autocomplete="list"
                      aria-controls="rc_select_TEST_OR_SSR_list"
                      aria-haspopup="listbox"
                      aria-label="Add-people-ariaLabel"
                      aria-owns="rc_select_TEST_OR_SSR_list"
                      autocomplete="off"
                      class="ant-select-selection-search-input"
                      id="rc_select_TEST_OR_SSR"
                      role="combobox"
                      type="search"
                      value=""
                    />
                    <span
                      aria-hidden="true"
                      class="ant-select-selection-search-mirror"
                    >
                       
                    </span>
                  </div>
                </div>
              </div>
              <span
                class="ant-select-selection-placeholder"
              >
                Add people and groups
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-col ant-col-4"
      >
        <div
          class="alignment"
        >
          <div
            class="ant-select condition-container-custom-select ant-select-single ant-select-show-arrow"
          >
            <div
              class="ant-select-selector"
            >
              <span
                class="ant-select-selection-search"
              >
                <input
                  aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                  aria-autocomplete="list"
                  aria-controls="rc_select_TEST_OR_SSR_list"
                  aria-haspopup="listbox"
                  aria-owns="rc_select_TEST_OR_SSR_list"
                  autocomplete="off"
                  class="ant-select-selection-search-input"
                  id="rc_select_TEST_OR_SSR"
                  readonly=""
                  role="combobox"
                  style="opacity: 0;"
                  type="search"
                  unselectable="on"
                  value=""
                />
              </span>
              <span
                class="ant-select-selection-item"
                title="Viewer"
              >
                Viewer
              </span>
            </div>
            <span
              aria-hidden="true"
              class="ant-select-arrow"
              style="user-select: none;"
              unselectable="on"
            >
              <span
                aria-label="down"
                class="anticon anticon-down ant-select-suffix"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="down"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                  />
                </svg>
              </span>
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      class="SelectedPeoples-container_checkbox"
    >
      <div>
        <label
          class="ant-checkbox-wrapper ant-checkbox-wrapper-checked"
        >
          <span
            class="ant-checkbox ant-checkbox-checked"
          >
            <input
              class="ant-checkbox-input"
              name="Notify people"
              type="checkbox"
              value=""
            />
            <span
              class="ant-checkbox-inner"
            />
          </span>
        </label>
        <span
          class="ant-typography"
          style="margin-left: 1rem;"
        >
          Notify people
        </span>
      </div>
      <div
        class="SelectedPeoples-container_textArea"
        id=""
      >
        <span
          class="ant-typography text-area-lable"
        >
          Message
        </span>
        <textarea
          class="ant-input text-area-input"
          name=""
          placeholder=""
        />
      </div>
    </div>
  </div>
</div>
`;
